utils/getLocalisedString.ts

```diff
@@ -8,3 +8,20 @@ export const getLocalizedString = (translations: LocalizedObject[] | undefined,
     }
     return translation.parts.map(p => (p as ExpressionArg).str).join('');
 }
+
+export const updateLocalizedString = (localizedString: LocalizedString[], langCode: string, value: string): LocalizedString[] => {
+    if (localizedString == undefined) {
+        return [{ code: langCode, parts: [{ str: value }] }];
+    }
+    const updated = [...localizedString];
+    const existing = updated.find(ls => ls.code === langCode);
+    if (existing) {
+        existing.parts = [{ str: value }];
+    } else {
+        updated.push({
+            code: langCode,
+            parts: [{ str: value }]
+        });
+    }
```

Comment:
I changed the method so it can also accept a list of strings. If we want to add more complexity, I would suggest to start using classes for these complex datatypes, and reorganise the code. I am wondering why classes are used so rarely anyways? Is there a specific reason? Why is everything an interface and nothing a class, when there are obvious needs for object specific functions (e.g. all the helper functions)? Calling localizedString.toString(lang: string) seems like a better way of dealing with things compared to getLocalizedString(x) defined in a separate file. Since transforming a LocalizedString into a string is completely dependant on how a LocalizedString works internally, the associated logic should rather be inside of the class / object itself and not separate. What do you think?